### PR TITLE
Asset doctype specification

### PR DESCRIPTION
**Pull request by BlackbitNeueMedien**
Some doctypes in Asset class only declared parameters of type `array`. As this makes auto-complete implossible this PR specifies the type of the array.

### CODE REVIEW

File: models/Asset.php [hunk 1]
```diff
@@ -1400,7 +1400,7 @@ public function getProperties()
     }
 
     /**
-     * @param array $properties
+     * @param Property[] $properties
```
Comment (weisswurstkanone) [inline models/Asset.php]: do we need a fix for Documents and DataObjects as well ?

File: models/Element/ValidationException.php [hunk 2]
```diff
@@ -38,7 +38,7 @@ public function getSubItems()
     /**
      * @param array $subItems
      */
-    public function setSubItems($subItems)
+    public function setSubItems(array $subItems)
```
Comment (dpfaffenbauer) [inline models/Element/ValidationException.php]: that actually is a BC break, even though I could live with it. Still...

File: models/Asset.php (hunk 1 above)
Comment (BlackbitNeueMedien) [reply models/Asset.php]: Done

File: models/Element/ValidationException.php (hunk 2 above)
Comment (BlackbitNeueMedien) [reply models/Element/ValidationException.php]: You are right. In my opinion this method gets mostly used internally (in fact I just wondered why I got `null` when calling `getSubItems` although its method signature says `@return array`.
But if the core team sees any difficulties in adding the typehint we can first add a deprecation hint...

Comment (weisswurstkanone) [reply models/Element/ValidationException.php]: Not sure about "mostly used internally" as the ValidationException class itself is mentioned in the docs.

https://pimcore.com/docs/5.x/Development_Documentation/Extending_Pimcore/Bundle_Developers_Guide/Plugin_Backend_UI.html

But yes, it is quite unlikely that the method is used quite often.

Comment (BlackbitNeueMedien) [reply models/Element/ValidationException.php]: I changed the code to only trigger a deprecation warning. We can introduce the typehint later...

**Pull request merged**